File: ClientApp/Dashboard/src/types/admin/engagement-analytics.ts
// Admin Engagement Analytics Types

export interface EngagementAnalytics {
  averageSessionDuration: number;
  bounceRate: number;
  pageViews: number;
  engagementRate: number;
  topPages: PageData[];
  engagementTrends: EngagementTrendData[];
}

export interface PageData {
  page: string;
  views: number;
}

export interface EngagementTrendData {
  date: string;
  sessions: number;
  pageViews: number;
  bounceRate: number;
}

File: src/Application/Features/Admin/Analytics/Models/Analytics/EngagementAnalytics.cs
using Application.Features.Admin.Analytics.Models.Trends;
using Application.Features.Admin.Analytics.Models.TopItems;

namespace Application.Features.Admin.Analytics.Models.Analytics
{
    public class EngagementAnalytics
    {
        public int TotalLikes { get; set; }
        public int TotalShares { get; set; }
        public int TotalViews { get; set; }
        public double EngagementRate { get; set; }
        public double AverageSessionDuration { get; set; }
        public List<EngagementTrendData> EngagementTrends { get; set; } = new();
        public List<TopEngagedContent> TopEngagedContent { get; set; } = new();
    }
}
